.github/scripts/check_pr_quality.py

```diff
@@ -1,112 +1,261 @@
 import os
 import re
 import sys
+from enum import Enum
 from github import Github
+from dataclasses import dataclass
+from typing import List
+
+
+class IssueType(Enum):
+    TUTORIAL = "tutorial"
+    MISSING_TICKET = "missing_ticket"
+    DJANGO_TUTORIAL = "django_tutorial"
+
+
+class Severity(Enum):
+    HIGH = "high"
+    MEDIUM = "medium"
+    LOW = "low"
+
+
+class Location(Enum):
+    TITLE = "title"
+    BODY = "body"
+
+
+class Confidence(Enum):
+    LOW = "low"
+    MEDIUM = "medium"
+    HIGH = "high"
+
+
+@dataclass
+class PatternMatch:
+    """Stores information about a matched pattern"""
+
+    pattern: str
+    matched_text: str
+    location: Location
+    context: str  # surrounding text
+
+
+@dataclass
+class QualityIssue:
+    """Represents a quality issue found in a PR"""
+
+    type: IssueType
+    severity: Severity
+    message: str
+    matches: List[PatternMatch]
+
+    def should_auto_close(self) -> bool:
+        """Determine if this issue warrants auto-closing"""
+        if self.type != IssueType.TUTORIAL:
+            return False
+
+        # Need at least one match to consider auto-closing
+        if not self.matches:
+            return False
+
+        # Auto-close if multiple matches
+        if len(self.matches) > 1:
+            return True
+
+        # Auto-close if single match is high-confidence pattern
+        pattern = self.matches[0].pattern
+        return pattern in ["first contribution", "first pr", "learning to"]
+
+
+class Action:
+    """Base class for all actions"""
+
+    def act(self, pr):
+        pass
+
+
+class NoAction(Action):
+    def act(self, pr):
+        print("No quality issues detected - PR looks good!")
+
+
+@dataclass
+class LabelPR(Action):
+    label: str
+
+    def act(self, pr):
+        pr.add_to_labels(self.label)
+
+
+@dataclass
+class CommentOnPR(Action):
+    comment: str
+
+    def act(self, pr):
+        print("Quality issues detected")
+        pr.create_issue_comment(self.comment)
+
+
+class ClosePR(Action):
+    def act(self, pr):
+        pr.edit(state="closed")
+
+
+def generate_actions(issues: List[QualityIssue]) -> List[Action]:
+    """Generate a list of actions to take based on issues found"""
+    if not issues:
+        return [NoAction()]
+
+    actions = [LabelPR("possibly-tutorial-pr")]
```

Comment:
This is only correct if one or more of the issues has a `type` that is `IssueType.TUTORIAL` (or maybe `IssueType.DJANGO_TUTORIAL`):

```suggestion
    actions = []
    if any(issue.type is IssueType.TUTORIAL for issue in issues):
        actions.append(LabelPR("possibly-tutorial-pr"))
```

or

```suggestion
    actions = []
    if any(issue.type in (IssueType.TUTORIAL, IssueType.DJANGO_TUTORIAL) for issue in issues):
        actions.append(LabelPR("possibly-tutorial-pr"))
```